tests/network/vmi_infosource.go

```diff
@@ -133,6 +133,14 @@ var _ = SIGDescribe("Infosource", func() {
 				return dummyInterfaceExists(vmi)
 			}, 120*time.Second, 2*time.Second).Should(Equal(true))
 
+			networkInterface := netvmispec.LookupInterfaceStatusByMac(vmi.Status.Interfaces, primaryInterfaceMac)
+			Expect(networkInterface).NotTo(BeNil(), "interface not found")
+			Expect(networkInterface.IP).To(BeEmpty())
+
+			guestInterface := netvmispec.LookupInterfaceStatusByMac(vmi.Status.Interfaces, primaryInterfaceNewMac)
+			Expect(guestInterface).NotTo(BeNil(), "interface not found")
+			Expect(guestInterface.IP).NotTo(BeEmpty())
+
 			for i := range vmi.Status.Interfaces {
```

Comment:
What is needed to remove this one out? It is after-all the source of all the above code.

Reply:
add cloud-init network data for eth0 and eth1 i believe so the ips will be predictable (or just ip commands in userData?)
but i think it is better that the test will be as simple as possible, it has enough logic in it already

imo this test checks infoSource and mac change, so the first expect would be enough (as we have enough tests that check ip of guest-agent reported interfaces)

the normalization was because we it make this test simpler and focus on what it actually tests
infoSource (and mac change on the way), i think it is better to keep it simple

Reply:
I think you can calculate everything based on the data in hand.
Said that, I have no interest in delaying this one from getting merged on this specific point.